src/main/java/org/terasology/dialogs/DialogScreen.java

```diff
@@ -71,12 +75,14 @@ public void reset() {
 
     public void addResponseOption(EntityRef charEnt, EntityRef talkTo, String text, List<PlayerAction> actions) {
         UIButton newButton = new UIButton();
+        UIImage newImage = new UIImage();
```

Comment:
That is pretty interesting! I'll have to look a bit more into how to use the responses, but it does seem like something I can figure out 👍 How terasology incorporates URNs is something I need to look more into as well, though I think there are plenty of places I can search to learn more.

Reply:
About having the image when set through the prefab, and having no image if no image is set at the prefab: what you mean is that if we do not set an image, then the dialogue works exactly like it does now, but if we add the `responseImage`, then a space which equals the width of the `responseImage` is added to the left of each answer button, right? (and when an answer is selected then the image will appear)

Reply:
Yes, either that, or the space is already there but no icon will appear there. @jdrueckert suggested to have a default (hover) image in dialogs to use when no icon is specified, this may be another option.

I'd imagine that the hover image will be scaled to fit the space available in the layout, which means that it might be scaled up or down accordingly.